platform-sdk/swirlds-state-impl/src/main/java/com/swirlds/state/merkle/disk/OnDiskWritableSingletonState.java

```diff
@@ -20,54 +21,64 @@
  * An implementation of {@link WritableSingletonState} backed by a {@link VirtualMap}, resulting in a state
  * that is stored on disk.
  *
- * @param <T> The type of the value
+ * @param <V> The type of the value
  */
-public class OnDiskWritableSingletonState<T> extends WritableSingletonStateBase<T> {
+public class OnDiskWritableSingletonState<V> extends WritableSingletonStateBase<V> {
```

Comment:
No we are getting rid of the in memory implementation, should we consider merging the interface and single implementation? On the principle of interfaces slow things down and just keep it simple, less classes less files all make life easier.

Reply:
Here it is: https://github.com/hiero-ledger/hiero-consensus-node/issues/21077